### ISSUE TITLE
GCNConv nan

### ISSUE THREAD

**Issue by AntoinePrv** [rusty1s/pytorch_geometric#61]
Hello,

I'm referring to the following found [here](deg_inv):
```python
deg = scatter_add(edge_attr, row, dim=0, dim_size=x.size(0))
deg_inv = deg.pow(-0.5)
deg_inv[deg_inv == float('inf')] = 0
```

I wonder why `deg` is supposed to represent a degree if arbitrary `edge_attr` are passed. Indeed, when taking the inverse square root, the result can be undefined.
In the third line, I get `nan`s rather than `inf`. They are not corrected and propagated in the rest of the code.

Wouldn't
```python
deg = scatter_add(torch.ones_like(edge_attr), row, dim=0, dim_size=x.size(0))
```
be the relevant degree ?

Thanks

**Comment by rusty1s** [comment]
Hi,
if edge weights are passed, then `deg` holds the weighted degree. This is equivalent to doing `A.sum(dim=1)` and corresponds to the computation in the [official repo](https://github.com/tkipf/gcn/blob/master/gcn/utils.py#L124).

I do not see how you can get `nan`s there. Do you have a short example for reproducing this issue?

**Comment by rusty1s** [comment]
Yes they need to be positive because of the square root.

**Comment by AntoinePrv** [comment]
Sorry for the misunderstanding and thank you for your time.
